bundle/permissions/mutator.go

```diff
@@ -0,0 +1,136 @@
+package permissions
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+)
+
+const CAN_MANAGE = "CAN_MANAGE"
+const CAN_VIEW = "CAN_VIEW"
+const CAN_RUN = "CAN_RUN"
+
+var allowedLevels = []string{"CAN_MANAGE", "CAN_VIEW", "CAN_RUN"}
+var levelsMap = map[string](map[string]string){
+	"jobs": {
+		CAN_MANAGE: "CAN_MANAGE",
+		CAN_VIEW:   "CAN_VIEW",
+		CAN_RUN:    "CAN_MANAGE_RUN",
+	},
+	"pipelines": {
+		CAN_MANAGE: "CAN_MANAGE",
+		CAN_VIEW:   "CAN_VIEW",
+		CAN_RUN:    "CAN_RUN",
+	},
+	"mlflow_experiments": {
+		CAN_MANAGE: "CAN_MANAGE",
+		CAN_VIEW:   "CAN_READ",
+	},
+	"mlflow_models": {
+		CAN_MANAGE: "CAN_MANAGE",
+		CAN_VIEW:   "CAN_READ",
+	},
+	"model_serving_endpoints": {
+		CAN_MANAGE: "CAN_MANAGE",
+		CAN_VIEW:   "CAN_VIEW",
+		CAN_RUN:    "CAN_QUERY",
+	},
+}
+
+type topLevelPermissions struct{}
+
+func ApplyTopLevelPermissions() bundle.Mutator {
+	return &topLevelPermissions{}
+}
+
+func (m *topLevelPermissions) Apply(ctx context.Context, b *bundle.Bundle) error {
+	err := m.validate(b)
+	if err != nil {
+		return err
+	}
+
+	m.applyForJobs(ctx, b)
+	m.applyForPipelines(ctx, b)
+	m.applyForMLModels(ctx, b)
+	m.applyForMLExperiments(ctx, b)
+	m.applyForModelServiceEndpoints(ctx, b)
+
+	return nil
+}
+
+func (m *topLevelPermissions) validate(b *bundle.Bundle) error {
+	for _, p := range b.Config.Permissions {
+		if !slices.Contains(allowedLevels, p.Level) {
+			return fmt.Errorf("invalid permission level: %s, allowed values: [%s]", p.Level, strings.Join(allowedLevels, ", "))
```

Comment:
When the location annotations work we can figure out exactly where this was defined and emit a diagnostic.